include/boost/multiprecision/detail/functions/pow.hpp

```diff
@@ -499,27 +500,29 @@ inline void eval_pow(T& result, const T& x, const T& a)
       return;
    }
 
+   using si_type = typename boost::multiprecision::detail::canonical<int, T>::type;
+   using fp_type = typename std::tuple_element<0, typename T::float_types>::type;
+
    if ((a.compare(si_type(1)) == 0) || (x.compare(si_type(1)) == 0))
    {
       result = x;
       return;
    }
-   if (a.compare(si_type(0)) == 0)
+
+   const int fpc_x { eval_fpclassify(x) };
+   const int fpc_a { eval_fpclassify(a) };
```

Comment:
I don't think fpc_a is used anywhere else?  If not then comparison to zero should be faster, but even better would be `eval_is_zero`.